gulpfile.js

```diff
@@ -0,0 +1,870 @@
+    });
+});
+
+gulp.task('instrumentForCoverage', ['build'], function(done) {
+    var jscoveragePath = path.join('Tools', 'jscoverage-0.5.1', 'jscoverage.exe');
+    var cmdLine = jscoveragePath + ' Source Instrumented --no-instrument=./ThirdParty';
+    child_process.exec(cmdLine, function(error, stdout, stderr) {
+        if (error) {
+            console.log(stderr);
+            return done(error);
+        }
+        console.log(stdout);
+        done();
+    });
+});
+
+gulp.task('jsHint', ['build'], function() {
+    return gulp.src(jsHintFiles)
+        .pipe(jshint.extract('auto'))
+        .pipe(jshint())
+        .pipe(jshint.reporter('jshint-stylish'));
+});
+
+gulp.task('jsHint-watch', function() {
+    gulp.watch(jsHintFiles).on('change', function(event) {
+        gulp.src(event.path)
+            .pipe(jshint.extract('auto'))
+            .pipe(jshint())
+            .pipe(jshint.reporter('jshint-stylish'));
+    });
+});
+
+gulp.task('makeZipFile', ['release'], function() {
+    var builtSrc = gulp.src([
+        'Build/Apps/**',
+        'Build/Cesium/**',
+        'Build/CesiumUnminified/**',
+        'Build/Documentation/**'
+    ], {
+        base : '.'
+    });
+
+    var staticSrc = gulp.src([
+        'Apps/**',
+        '!Apps/Sandcastle/gallery/development/**',
+        'Source/**',
+        'Specs/**',
+        'ThirdParty/**',
+        'logo.png',
+        'favicon.ico',
+        'gulpfile.js',
+        'server.js',
+        'package.json',
+        'LICENSE.md',
+        'CHANGES.md',
+        'README.md',
+        'web.config'
+    ], {
+        base : '.',
+        nodir : true
+    });
+
+    var indexSrc = gulp.src('index.release.html').pipe(gulpRename("index.html"));
+
+    return eventStream.merge(builtSrc, staticSrc, indexSrc)
+        .pipe(gulpZip('Cesium-' + version + '.zip'))
+        .pipe(gulp.dest('.'));
+});
+
+gulp.task('minify', ['generateStubs'], function() {
+    return combineJavaScript({
+        removePragmas : false,
+        optimizer : 'uglify2',
+        outputDirectory : path.join('Build', 'Cesium')
+    });
+});
+
+gulp.task('minifyRelease', ['generateStubs'], function() {
+    return combineJavaScript({
+        removePragmas : true,
+        optimizer : 'uglify2',
+        outputDirectory : path.join('Build', 'Cesium')
+    });
+});
+
+gulp.task('release', ['buildApps', 'generateDocumentation']);
+
+gulp.task('generateStubs', ['build'], function(done) {
+    mkdirp.sync(path.join('Build', 'Stubs'));
+
+    var contents = '\
+/*global define,Cesium*/\n\
+(function() {\n\
+"use strict";\n\
+/*jshint sub:true*/\n';
+    var modulePathMappings = [];
+
+    globby.sync(sourceFiles).forEach(function(file) {
+        file = path.relative('Source', file);
+        var moduleId = filePathToModuleId(file);
+
+        var baseName = path.basename(file);
+        var propertyName = baseName.substring(0, baseName.lastIndexOf('.'));
+        propertyName = "['" + String(propertyName) + "']";
```

Comment:
the cast to String is leftover from the Rhino implementation and should no longer be necessary.